### ISSUE TITLE
.dropna call fails with pandas 1.5.0

### ISSUE THREAD

**Issue by thehomebrewnerd** [dask/dask#9503]
**What happened**:
Calling `.dropna` on a Dask dataframe results in an error with pandas 1.5.0, but works fine with pandas 1.4.4.

**What you expected to happen**:
`.dropna` call should work with pandas 1.5.0 installed.

**Minimal Complete Verifiable Example**:

```python
import pandas as pd
import dask.dataframe as dd

df = pd.DataFrame({"id": [0, 1, 2], "vals": [0, pd.NA, 2]})
dask_df = dd.from_pandas(df, npartitions=1)
dask_df.dropna()
```

**Anything else we need to know?**:
With pandas 1.4.4, the `.dropna` call results in the expected dataframe:
```
   id vals
0   0    0
2   2    2
```

With pandas 1.5.0, the following error occurs:
```
ValueError: Metadata inference failed in `dropna`.
```

**Environment**:

- Dask version: 2022.7.1
- Python version: 3.8.10
- Operating System: osx
- Install method (conda, pip, source): pip

**Comment by jrbourbeau** [comment]
Thanks for reporting @thehomebrewnerd. I'm able to reproduce with `pandas=1.5.0` and `dask=2022.7.1`, but the issue is resolved when I use a more recent version of `dask` (specifically using the latest `dask=2022.9.1` release locally). Could you try upgrading `dask` to see if that resolves your issue? The `2022.9.1` release contains several `pandas=1.5` compatibility fixes.

**Comment by thehomebrewnerd** [comment]
@jrbourbeau Yes, upgrading fixes for me as well. I thought I had grabbed the most recent Dask version, but didn't realize one of our libraries was restricting the Dask version to `<2022.8.1` until just now. Will try to resolve that separately and will close this issue.

Thanks for the quick response!
